ui/src/components/ui/device-control-panel/tabs/app-upload-control/app-upload-control.tsx

```diff
@@ -0,0 +1,60 @@
+import { useState } from 'react'
+import { observer } from 'mobx-react-lite'
+import { useTranslation } from 'react-i18next'
+import { FormStatus, Spacing } from '@vkontakte/vkui'
+
+import { FileInput } from '@/components/lib/file-input'
+import { ProgressBar } from '@/components/lib/progress-bar'
+import { ConditionalRender } from '@/components/lib/conditional-render'
+
+import { ApplicationInstallationService } from '@/services/application-installation/application-installation-service'
+
+import { useServiceLocator } from '@/lib/hooks/use-service-locator.hook'
+import { useDeviceSerial } from '@/lib/hooks/use-device-serial.hook'
+
+import { ActivityLauncher } from './activity-launcher'
+
+export const AppUploadControl = observer(() => {
+  const { t } = useTranslation()
+  const serial = useDeviceSerial()
+  const [fileInputError, setFileInputError] = useState('')
+  const applicationInstallationService = useServiceLocator<ApplicationInstallationService>(
+    ApplicationInstallationService.name
+  )
+
+  return (
+    <>
+      <FileInput
+        accept={[
+          '.apk',
+          '.aab',
+          '.ipa',
```

Comment:
We can upload .ipa only for iOS devices (but it is not working yet)

Reply:
Should I remove .ipa from this array for now?

Reply:
I think we dont need to accept ipa's on android devices